contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -3528,6 +3536,105 @@ PlTsqlMatchNamedCall(HeapTuple proctup, int nargs, List *argnames,
 	return true;
 }
 
+static int getDefaultPosition(const List *default_positions, const ListCell *def_idx, int argPosition)
+{
+	int currPosition = intVal((Node *) lfirst(def_idx));
+	while (currPosition != argPosition)
+	{
+		def_idx = lnext(default_positions, def_idx);
+		if (def_idx == NULL)
+		{
+			elog(ERROR, "not enough default arguments");
+			return -1;
+		}
+		currPosition = intVal((Node *) lfirst(def_idx));
+	}
+	return list_cell_number(default_positions, def_idx);
+}
+
+/**
+ * @brief farg position default should get the corresponding default position value
+ * 
+ * @param func_tuple 
+ * @param defaults 
+ * @param fargs 
+ * @return List* 
+ */
+static List*
+replace_pltsql_function_defaults(HeapTuple func_tuple, List *defaults, List *fargs)
+
+{
+	HeapTuple	bbffunctuple;
+
+	if (sql_dialect != SQL_DIALECT_TSQL)
+		return fargs;
+
+	bbffunctuple = get_bbf_function_tuple_from_proctuple(func_tuple);
```

Comment:
`get_bbf_function_tuple_from_proctuple` returns `NULL` if `babelfish_dump_restore` is true.
ie: In the case when `babelfish_dump_restore` is true then this function will error out.

Reply:
yeah that make sense